nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/AbstractAzureDataLakeStorageProcessor.java

```diff
@@ -146,8 +154,22 @@ public static DataLakeServiceClient getStorageClient(PropertyContext context, Fl
         return storageClient;
     }
 
-    @Override
-    public Set<Relationship> getRelationships() {
-        return RELATIONSHIPS;
+    private static class DirectoryValidator implements Validator {
+        @Override
+        public ValidationResult validate(String subject, String input, ValidationContext context) {
+            ValidationResult.Builder builder = new ValidationResult.Builder()
+                    .subject(DIRECTORY.getDisplayName())
+                    .input(input);
+
+            if (context.isExpressionLanguagePresent(input)) {
+                builder.valid(true).explanation("Expression Language Present");
+            } else if (input.startsWith("/")) {
+                builder.valid(false).explanation(String.format("'%s' cannot contain a leading '/'", DIRECTORY.getDisplayName()));
+            } else {
+                builder.valid(true);
```

Comment:
I think this should be `builder.valid(StringUtils.nonBlank(input))` or similar to catch it early. If you run this with non-zero length whitespace you'll get an error at run time.

Reply:
Good point, we can catch the whitespace-only value here. Will add it.